From 84f63b6d20863749b746031a9d14949f8f3051d8 Mon Sep 17 00:00:00 2001
From: Dan Douglass <dan.douglass@freescale.com>
Date: Tue, 26 May 2015 15:59:39 -0500
Subject: [PATCH 1567/1594] MLK-10978-2 ARM: dts: Add CAAM support for i.mx6ul

commit b92f3487cbcda1c4f458f3456e9f0760b69091df from
git://git.freescale.com/imx/linux-2.6-imx.git

Add CAAM device tree node for i.mx6ul.

Signed-off-by: Dan Douglass <dan.douglass@freescale.com>
---
 arch/arm/boot/dts/imx6ul.dtsi |   24 ++++++++++++------------
 1 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ul.dtsi b/arch/arm/boot/dts/imx6ul.dtsi
index 745d363..fed7975 100644
--- a/arch/arm/boot/dts/imx6ul.dtsi
+++ b/arch/arm/boot/dts/imx6ul.dtsi
@@ -173,6 +173,11 @@
 			reg = <0x00905000 0x1B000>;
 		};
 
+		caam_sm: caam-sm@00100000 {
+			 compatible = "fsl,imx7d-caam-sm", "fsl,imx6q-caam-sm";
+			 reg = <0x00100000 0x3fff>;
+		};
+
 		aips1: aips-bus@02000000 {
 			compatible = "fsl,aips-bus", "simple-bus";
 			#address-cells = <1>;
@@ -778,31 +783,26 @@
 			ranges;
 
 			crypto: caam@2140000 {
-				compatible = "fsl,sec-v4.0";
+				compatible = "fsl,imx6ul-caam", "fsl,sec-v4.0";
 				#address-cells = <1>;
 				#size-cells = <1>;
 				reg = <0x2140000 0x3c000>;
 				ranges = <0 0x2140000 0x3c000>;
-				interrupt-parent = <&intc>; /* interrupts = <0 92 0x4>; */
-				clocks = <&clks IMX6UL_CLK_DUMMY>, <&clks IMX6UL_CLK_DUMMY>,
-					 <&clks IMX6UL_CLK_DUMMY> ,<&clks IMX6UL_CLK_DUMMY>;
-				clock-names = "caam_mem", "caam_aclk", "caam_ipg", "caam_emi_slow";
-				status = "disabled";
+				interrupts = <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clks IMX6UL_CLK_CAAM_IPG>, <&clks IMX6UL_CLK_CAAM_ACLK>,
+				       	 <&clks IMX6UL_CLK_CAAM_MEM>;
+				clock-names = "caam_ipg", "caam_aclk", "caam_mem";
 
 				sec_jr0: jr0@1000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x1000 0x1000>;
-					interrupt-parent = <&intc>;
-					interrupts = <0 105 0x4>;
-					status = "disabled";
+					interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
 				};
 
 				sec_jr1: jr1@2000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x2000 0x1000>;
-					interrupt-parent = <&intc>;
-					interrupts = <0 106 0x4>;
-					status = "disabled";
+					interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>;
 				};
 			};
 
-- 
1.7.5.4

